src/main/java/Problem2/LCA.java

```diff
@@ -4,8 +4,23 @@
 
 public class LCA {
     public static TreeNode<Integer> lowestCommonAncestor(TreeNode<Integer> root, TreeNode<Integer> p, TreeNode<Integer> q) {
-        // homework
-        // hint: pay attention to the assumptions of this problem.
-        return null;    // place holder
+        if (root == null) {
+            return null;
+        }
+        if (root == p || root == q) {
+            return root;
+        }
+        TreeNode<Integer> left = lowestCommonAncestor(root.left, p, q);
+        TreeNode<Integer> right = lowestCommonAncestor(root.right, p, q);
+
+        if (left != null && right != null) {
+            return root;
+        }
+
+        if (right != null) {
+            return right;
+        } else {
+            return left;
+        }
```

Comment:
What abut when both == NULL?

Reply:
If any of the child is null it will return null, because it's a recursively calling itself from line 13 and 14, and line 7 is checking for null, and returning null

Reply:
Do you have any additional tests in mind that I missed?